### PR TITLE
[DUOS-1461][risk=no] Restrict chair edit for datasets to dac-owned ones

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1461

See also https://github.com/DataBiosphere/duos-ui/pull/1295 for front end changes

* Add validation to the edit, disable, and delete dataset endpoints for chairpersons. 
* Ensure that the dataset is one that exists on one of the chairperson's dacs.
* Minor updates to ensure we are populating the deletable property for datasets
* Minor query refactoring/formatting for efficiency

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java [hunk 1]
```diff
@@ -346,4 +359,23 @@ protected Logger logger() {
...
+        List<Integer> dacIds = user.getRoles().stream()
+            .filter(r -> r.getRoleId().equals(UserRoles.CHAIRPERSON.getRoleId()))
+            .map(UserRole::getDacId)
+            .collect(Collectors.toList());
+        if (dacIds.isEmpty()) {
+            // Something went very wrong here. A chairperson with no dac ids is an error
+            logger().error("Unable to find dac ids for chairperson user: " + user.getEmail());
+            throw new NotFoundException();
+        } else {
+            Set<DataSet> accessibleDatasets = dacService.findDatasetsByDacIds(dacIds);
```
Comment (JVThomas) [inline src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java]: Would it be faster to just get the dataset's Consent record and see if the dacId on that matches the Chair's dacIds? `ConsentService.getConsentFromDatasetID` can help with that.

**Review by JVThomas — APPROVED** [review]
Code works as expected, just have one comment on validateDatasetDacAccess

File: src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java]: That does seem like a simpler approach. Will look at it!

**Comment by rushtong** [comment]
@JVThomas FYI, I made some non-trivial changes for your suggestion, PTAL, thank you!

**Review by JVThomas — APPROVED** [review]
Looks good. Tested locally, Chairs cannot edit datasets outside of their DAC.

**Pull request merged**